buildSrc/src/main/minimumRuntime/org/elasticsearch/gradle/LoggedExec.java

```diff
@@ -19,21 +26,52 @@
 @SuppressWarnings("unchecked")
 public class LoggedExec extends Exec {
 
+    // helper to allow lambda creation using Files methods
+    private interface ThrowingStringSupplier {
+        String get() throws Exception;
+    }
+
+    private Property<Boolean> useFileBuffer;
+    
     public LoggedExec() {
-        ByteArrayOutputStream output = new ByteArrayOutputStream();
-        ByteArrayOutputStream error = new ByteArrayOutputStream();
+        this.useFileBuffer = getProject().getObjects().property(Boolean.class);
+        this.useFileBuffer.set(false); // default to in memory
         if (getLogger().isInfoEnabled() == false) {
-            setStandardOutput(output);
-            setErrorOutput(error);
-            setIgnoreExitValue(true);
-            doLast((unused) -> {
+            getProject().afterEvaluate(p -> {
+                OutputStream stdout;
+                OutputStream stderr;
+                ThrowingStringSupplier readStdout;
+                ThrowingStringSupplier readStderr;
+
+                try {
+                    if (useFileBuffer.get()) {
+                        String outputdir = p.getBuildDir() + "/buffered-output/" + this.getName();
+                        p.mkdir(outputdir);
```

Comment:
FWIW, if you declare a property annotated with `@OutputDirectory`, Gradle will take care of creating it for you. Not strictly necessary here since we don't want this task to be incremental, just a helpful tip.